uk.co.compucorp.civicrm.hrleaveandabsences/views/shared/components/leave-calendar-day.html

```diff
@@ -1,31 +1,37 @@
 <div class="chr_leave-calendar__day-container"
   uib-tooltip-template="'chr_leave-calendar__day-tooltip'"
-  tooltip-append-to-body="false"
+  tooltip-animation="false"
+  tooltip-class="chr_leave-calendar__day-tooltip"
+  tooltip-is-open="day.tooltip.show"
+  tooltip-append-to-body="true"
+  tooltip-trigger="none"
   tooltip-enable="day.contactData.leaveRequests.length"
+  ng-mouseenter="day.toggleTooltip('day_cell', true)"
+  ng-mouseleave="day.toggleTooltip('day_cell', false)"
   ng-class="{
     'chr_leave-calendar__day-container--type--weekend': day.contactData.isWeekend,
     'chr_leave-calendar__day-container--non-working': day.contactData.isNonWorkingDay,
     'chr_leave-calendar__day-container--public-holiday': day.contactData.isPublicHoliday
   }">
   <div class="chr_leave-calendar__item"
     ng-repeat="(leaveRequestIndex, leaveRequest) in day.contactData.leaveRequestsToShowInCell"
-    ng-init="leaveRequestAttributes = day.contactData.leaveRequestsAttributes[leaveRequest.id]"
-    ng-style="leaveRequestAttributes.styles"
+    ng-style="day.contactData.leaveRequestsAttributes[leaveRequest.id].styles"
     ng-hide="day.contactData.leaveRequests.length >= 3 && leaveRequestIndex > 0"
     ng-class="{
       'chr_leave-calendar__item--half': day.contactData.leaveRequestsToShowInCell.length >= 2,
-      'chr_leave-calendar__item--requested': leaveRequestAttributes.isRequested,
-      'chr_leave-calendar__item--alt': leaveRequestAttributes.isAccruedTOIL
+      'chr_leave-calendar__item--requested': day.contactData.leaveRequestsAttributes[leaveRequest.id].isRequested,
+      'chr_leave-calendar__item--alt': day.contactData.leaveRequestsAttributes[leaveRequest.id].isAccruedTOIL
     }">
     <span ng-if="leaveRequest">
       <span class="chr_leave-calendar__item__label">
-        {{leaveRequestAttributes.label}}
+        {{day.contactData.leaveRequestsAttributes[leaveRequest.id].label}}
```

Comment:
it would be nice if we had some sort of object that contains the `leaveRequest` and the `leaveRequestAttributes` so we could do something like

```html
<span class="chr_leave-calendar__item__label">
  {{leaveRequest.attributes.label}}
</span>
```

But again, this is just a suggestion, if it's too complicated, leave it.

Reply:
This is not complicated, the reason is a bit more strong - we do not want to contaminate `leaveRequest` object since it is then passed to other components such as Leave Request Modal. I am not personally a fan of passing the whole object, I would rather pass just ID, but this is how it is done and, I guess, for performance (no need to load leave request again). So we don't really want to change the way it works now and do not want to contaminate LR object at the same time. I agree, this is not really beautiful in the code, I tried to use `ng-init` to store the reference, but `ng-init` does not update later if we modify props of the referenced object, thus UI did not update as well.